src/array.ts

```diff
@@ -29,6 +29,7 @@ declare global {
     each(f: (el: T, index: number, array: T[]) => any): T[];
     /** Helper for filtering arrays on async predicates. */
     asyncFilter(predicate: (v: T) => Promise<boolean>): Promise<Array<T>>;
```

Comment:
I feel changing the return type of `asyncFilter`'s `predicate` to `boolean | Promise<boolean>` and the return type of `asyncMap`'s `f` to `V | Promise<V>` can allow for more flexibility and potential performance benefit. If desired, I can make tweaks to those methods in this PR or a separate PR.